build/parseSpec.c

```diff
@@ -400,71 +449,74 @@ int readLine(rpmSpec spec, int strip)
     SKIPSPACE(s);
 
     match = -1;
-    if (!spec->readStack->reading && ISMACROWITHARG(s, "%if")) {
-	match = 0;
-    } else if (ISMACROWITHARG(s, "%ifarch")) {
-	ARGMATCH(s, "%{_target_cpu}", match);
-    } else if (ISMACROWITHARG(s, "%ifnarch")) {
-	ARGMATCH(s, "%{_target_cpu}", match);
-	match = !match;
-    } else if (ISMACROWITHARG(s, "%ifos")) {
-	ARGMATCH(s, "%{_target_os}", match);
-    } else if (ISMACROWITHARG(s, "%ifnos")) {
-	ARGMATCH(s, "%{_target_os}", match);
-	match = !match;
-    } else if (ISMACROWITHARG(s, "%if")) {
-	s += 3;
-        match = parseExpressionBoolean(s);
-	if (match < 0) {
-	    rpmlog(RPMLOG_ERR,
-			_("%s:%d: bad %%if condition\n"),
-			ofi->fileName, ofi->lineNum);
-	    return PART_ERROR;
-	}
-    } else if (ISMACRO(s, "%else")) {
-	if (! spec->readStack->next) {
-	    /* Got an else with no %if ! */
-	    rpmlog(RPMLOG_ERR,
-			_("%s:%d: Got a %%else with no %%if\n"),
-			ofi->fileName, ofi->lineNum);
-	    return PART_ERROR;
-	}
-	spec->readStack->reading =
-	    spec->readStack->next->reading && ! spec->readStack->reading;
-	spec->line[0] = '\0';
-    } else if (ISMACRO(s, "%endif")) {
-	if (! spec->readStack->next) {
-	    /* Got an end with no %if ! */
-	    rpmlog(RPMLOG_ERR,
-			_("%s:%d: Got a %%endif with no %%if\n"),
-			ofi->fileName, ofi->lineNum);
-	    return PART_ERROR;
-	}
-	rl = spec->readStack;
-	spec->readStack = spec->readStack->next;
-	free(rl);
-	spec->line[0] = '\0';
-    } else if (spec->readStack->reading && ISMACROWITHARG(s, "%include")) {
-	char *fileName, *endFileName, *p;
-
-	fileName = s+8;
-	SKIPSPACE(fileName);
-	endFileName = fileName;
-	do {
-	    SKIPNONSPACE(endFileName);
-	    p = endFileName;
-	    SKIPSPACE(p);
-	    if (*p != '\0') endFileName = p;
-	} while (*p != '\0');
-	if (*fileName == '\0') {
-	    rpmlog(RPMLOG_ERR, _("%s:%d: malformed %%include statement\n"),
-				ofi->fileName, ofi->lineNum);
-	    return PART_ERROR;
-	}
-	*endFileName = '\0';
 
-	ofi = pushOFI(spec, fileName);
-	goto retry;
+    if ((lineType = parseLineType(s))) {
+	if (!spec->readStack->reading && (lineType->id == LINE_IF)) {
```

Comment:
Please jump to a goto label at the end instead of indenting the whole block. It's impossible to see what *really* changes when done like this.